app/services/fdc_backorderer.rb

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+
+# Place and update orders based on missing stock.
+class FdcBackorderer
+  attr_reader :user, :urls
+
+  def initialize(user, urls)
+    @user = user
+    @urls = urls
+  end
+
+  def find_or_build_order(ofn_order)
+    find_open_order || build_new_order(ofn_order)
+  end
+
+  def build_new_order(ofn_order)
+    OrderBuilder.new_order(ofn_order, urls.orders_url).tap do |order|
+      order.saleSession = build_sale_session(ofn_order)
+    end
+  end
+
+  def find_open_order
+    graph = import(urls.orders_url)
+    open_orders = graph&.select do |o|
+      o.semanticType == "dfc-b:Order" && o.orderStatus[:path] == "Held"
+    end
+
+    return if open_orders.blank?
+
+    # If there are multiple open orders, we don't know which one to choose.
+    # We want the order we placed for the same distributor in the same order
+    # cycle before. So here are some assumptions for this to work:
+    #
+    # * We see only orders for our distributor. The endpoint URL contains the
+    #   the distributor name and is currently hardcoded.
+    # * There's only one open order cycle at a time. Otherwise we may select
+    #   an order of an old order cycle.
+    # * Orders are finalised when the order cycle closes. So _Held_ orders
+    #   always belong to an open order cycle.
+    # * We see only our own orders. This assumption is wrong. The Shopify
+    #   integration places held orders as well and they are visible to us.
+    #
+    # Unfortunately, the endpoint doesn't tell who placed the order.
+    # TODO: We need to remember the link to the order locally.
+    # Or the API is updated to include the orderer.
+    #
+    # For now, we just guess:
+    open_orders.last.tap do |order|
+      # The DFC Connector doesn't recognise status values properly yet.
```

Comment:
@mkllnk could you elaborate on what you mean by "DFC Connector doesn't recognise status values properly"? 😟 

I thought this was sorted & we haven't had any issues with the TS Connector. @lecoqlibre do we need to update the Ruby Connector?

Reply:
Well, we can load the vocabulary with order states. The Connector doesn't have a built-in method for that but I did make it work in a separate branch. I could work on a pull request to make this part work with the importer. I was just shying away from the effort to meet the deadline. The current version is also not recognising the transformation links. That may just be the outdated context though. We are still on 1.8, I think. And I wasn't keen to follow the updates closely because you discovered bugs here and there. Happy to give it another go though.